### ISSUE TITLE
Vault UI: display error if OIDC redirect_url not set up correctly or invalid role selected

### ISSUE THREAD

**Issue by candlerb** [hashicorp/vault#11071]
**Is your feature request related to a problem? Please describe.**
When there is a problem with OIDC configuration, clicking on the "Sign in with OIDC provider" button is silently ignored.  This makes it hard to work out what is wrong, both from the system administrator's point of view setting this up, and supporting end-users who are having difficulty logging in.

I see this when the auth role `allowed_redirect_uris` doesn't match the `redirect_uri` which the Vault UI sends; and I also see it when the user enters an auth role name which doesn't exist.

### Reproducer case 1
Write an auth role where `allowed_redirect_uris` doesn't correctly match the redirect_uri which Vault UI will send, e.g. I put "myerror" instead of "google" in the path below.

```
vault write auth/google/role/standard - <<EOF
{
  "allowed_redirect_uris": ["https://mydomain.com:8200/ui/vault/auth/myerror/oidc/callback","http://localhost:8250/oidc/callback"],
  "user_claim": "sub",
  "oidc_scopes": ["profile","email"],
  "claim_mappings": {
    "name": "name",
    "nickname": "nickname",
    "email": "email"
  }
}
```

Make it visible in the UI:

```
vault auth tune -listing-visibility=unauth -description="Google Account" google/
```

Go to login page, click "Sign in with OIDC provider" button: nothing happens (and no error is displayed). 

![image](https://user-images.githubusercontent.com/44789/110627032-633f3c00-8199-11eb-8743-b76cb4ef9ae8.png)

Note: if you use the browser console to look at network activity, you can see a _successful_ access to `/v1/auth/google/oidc/auth_url` prior to clicking the button, but the response contains `{"auth_url":""}`.   This happens whether you leave the role as empty, or explicitly enter the full name of the default role (which is "standard" in the above example)

### Reproducer case 2

Enter an invalid role name:

![image](https://user-images.githubusercontent.com/44789/110626473-a8af3980-8198-11eb-8a34-e93ab0b40bb3.png)

Again, clicking on the "Sign in with OIDC provider" does nothing and gives no error.

Note: in this case, the browser console shows a 400 _error_ response from `/v1/auth/google/oidc/auth_url`, with body `{"errors":["role \"asdf\" could not be found"]}`.  That is at least helpful for the administrator to debug the issue, if they know how to use the browser's developer tools.  But it doesn't help the end-user.

**Describe the solution you'd like**
I these cases I would like a message to be displayed somewhere on the page: either at the moment where auth_url is fetched, or at the point when the button is clicked. e.g.

* _auth_url is empty_
* _role "asdf" could not be found_

**Describe alternatives you've considered**
N/A

**Explain any additional use-cases**
N/A

**Additional context**
When the `redirect_uri` is not in` allowed_redirect_uris`, the API `/v1/auth/<path>/oidc/auth_url` doesn't return an error. Rather, it returns a 200 response with `{"auth_url":""}`

```
$ curl -v -d role=standard -d redirect_uri=http://dummy $VAULT_ADDR/v1/auth/google/oidc/auth_url
...
< HTTP/2 200
< cache-control: no-store
< content-type: application/json
< content-length: 173
< date: Wed, 10 Mar 2021 11:38:52 GMT
<
{"request_id":"1f28917c-4935-1bf5-1e27-e707876aa570","lease_id":"","renewable":false,"lease_duration":0,"data":{"auth_url":""},"wrap_info":null,"warnings":null,"auth":null}
```

I wasn't expecting this: I think that a 400 response with `{"errors":["redirect_uri is not in allowed_redirect_uris"]}` would be better.  However that may break API backwards compatibility.  In that case, setting the `warnings` member may be better (currently it has `"warnings":null`)

**Comment by erickufrin** [comment]
Yep. Hard to know whats going on without direct access to the logs.

**Comment by candlerb** [comment]
As described above, you'll get an empty `auth_url` response when the `redirect_uri` that you provide is not in the `allowed_redirect_uris` for the role.

You need to provide both a `redirect_uri` and `role` in the POST body to the `auth_url` endpoint - see [API docs](https://www.vaultproject.io/api-docs/auth/jwt#oidc-authorization-url-request).

If that's not what you're seeing, then you'll have to describe how to reproduce the issue.  But I suspect the problem is that you're not constructing the correct redirect_uri in the role, to match your server's external URI.

**Comment by redsux** [comment]
Thanks for your answers @candlerb  !

It helps find what's missing in my `allowed_redirect_uris` (forgot a mount path).
And it was also missing in the Azure Application web redirect part ...

Sorry for the post !

**Comment by hellobontempo** [comment]
Hello there! Thank you for the feature suggestion 😄 I believe this was added as part of a bug fix in #17661
